pkg/gadgets/run/tracer/run.go

```diff
@@ -272,10 +272,49 @@ func (g *GadgetDesc) CustomParser(params *params.Params, args []string) (parser.
 	if err := cols.AddFields(fields, base); err != nil {
 		return nil, fmt.Errorf("adding fields: %w", err)
 	}
+	return cols, nil
+}
 
+func (g *GadgetDesc) CustomParser(params *params.Params, args []string) (parser.Parser, error) {
+	cols, err := g.getColumns(params, args)
+	if err != nil {
+		return nil, err
+	}
 	return parser.NewParser[types.Event](cols), nil
 }
 
+func (g *GadgetDesc) CustomJsonParser(params *params.Params, args []string, options ...columns_json.Option) (*columns_json.Formatter[types.Event], error) {
+	cols, err := g.getColumns(params, args)
+	if err != nil {
+		return nil, err
+	}
+	return columns_json.NewFormatter(cols.ColumnMap, options...), nil
+}
+
+func (g *GadgetDesc) JSONConverter(params *params.Params, printer gadgets.Printer) func(ev any) {
+	formatter, err := g.CustomJsonParser(params, []string{})
+	if err != nil {
+		printer.Logf(logger.WarnLevel, "creating json formatter: %s", err)
```

Comment:
mmm, I think we should find a way to propagate these errors, but we can handle it in a following PR.